### PR TITLE
Feature ui detail layout

### PR DESCRIPTION
**Pull request by SheldonCheng**
Some work for mobile view:
- Hide/reveal header according to scroll direction
- On post detail screen, Header is always shown, with Back button

### CODE REVIEW

**Comment by subhransu** [comment]
@SheldonCheng can you help to rebase this PR again .... there are some merge conflicts after #77 got merged.

File: ui/src/shared/components/AppContext.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import * as React from "react";
+
+type ContextProps = {
+  isDetailScreen?: boolean;
+  setIsDetailScreen?: React.Dispatch<boolean>;
+};
+
+const AppContext = React.createContext<ContextProps>({});
+
+export const AppContextProvider: React.FC = function ({ children }) {
```
Comment (lxcid) [inline ui/src/shared/components/AppContext.tsx]: On post details screen, we can use the url to determine if we are in details screens

e.g.

```ts
const match = useRouteMatch("/posts/:postId");
if (match == null) {
  // not detail screen
} else {
  const postId = match.params.postId;
  // detail screen
}
```

Comment (SheldonCheng) [reply ui/src/shared/components/AppContext.tsx]: @lxcid Thanks for the tip, completely forgot about that. I've removed AppContext and used useRouteMatch instead.

File: ui/src/components/Header/NavBack.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const NavBack = () => {
+  const {
+    location: { pathname },
+  } = useHistory();
+  const split = pathname.split("/");
```
Comment (lxcid) [inline ui/src/components/Header/NavBack.tsx:22]: Nit: Might not be the same behaviour that you want (e.g. if people navigate from other sites other than `/posts`) but there's a `goBack()` function in `history` that return you to previous page.

https://reacttraining.com/react-router/web/api/history

**Review by lxcid — APPROVED** [review]
LGTM!

File: ui/src/shared/components/AppContext.tsx (hunk 1 above)
Comment (lxcid) [reply ui/src/shared/components/AppContext.tsx]: np! glad it help! :D

**Pull request merged**